src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1869,6 +1869,21 @@ private void ImportCall(ILOpcode opcode, int token)
                 }
             }
 
+            // Hard coded InternalCall mappings for mono interoperability
+            if (callee.IsInternalCall)
+            {
+                var ecmaType = callee.OwningType as EcmaType;
+                // See https://github.com/dotnet/runtime/blob/9ba9a300a08170c8170ea52981810f41fad68cf0/src/mono/wasm/runtime/driver.c#L400-L407
+                // Mono have these InternalCall methods in different namespaces but just mapping them to CoreRT.WebAssembly.MonoBridge.
+                if (ecmaType != null && (ecmaType.Namespace == "WebAssembly.JSInterop" && ecmaType.Name == "InternalCalls" || ecmaType.Namespace == "WebAssembly" && ecmaType.Name == "Runtime"))
+                {
+                    var coreRtJsInternalCallsType = _compilation.TypeSystemContext
+                        .GetModuleForSimpleName("CoreRT.WebAssembly.MonoBridge")
+                        .GetKnownType("CoreRT.WebAssembly.MonoBridge", "InternalCalls");
+                    callee = coreRtJsInternalCallsType.GetMethod(callee.Name, null);
```

Comment:
```suggestion
                    callee = coreRtJsInternalCallsType.GetMethod(callee.Name, callee.Signature);
```

Reply:
Bah, never mind. I just looked what Mono is doing and the sig doesn't matter to them.

I guess messing up the sig will lead to fun debugging time.

Reply:
Seems like the right thing to do, will change.